src/routes/transactions/mappers/transactions-history.mapper.ts

```diff
@@ -188,54 +188,41 @@ export class TransactionsHistoryMapper {
     onlyTrusted: boolean,
   ): Promise<TransactionItem[]> {
     const limitedTransfers = transfers.slice(0, this.maxNestedTransfers);
-    const result: TransactionItem[] = [];
 
-    for (const transfer of limitedTransfers) {
-      const nestedTransaction = await this.transferMapper.mapTransfer(
-        chainId,
-        transfer,
-        safe,
-      );
-
-      const transferWithValue = this.mapZeroValueTransfer(nestedTransaction);
-      // If we do not have a transfer with value, we do not add it to the result
-      if (!transferWithValue) continue;
+    const nestedTransactions = await Promise.all(
+      limitedTransfers.map((transfer) =>
+        this.transferMapper.mapTransfer(chainId, transfer, safe),
+      ),
+    );
 
-      const trustedTransfer = onlyTrusted
-        ? this.mapTrustedTransfer(transferWithValue)
-        : transferWithValue;
+    return nestedTransactions
+      .filter((nestedTransaction): boolean => {
+        // If we do not have a transfer with value, we do not add it to the result
+        if (!this.isTransferWithVaklue(nestedTransaction)) return false;
 
-      if (!trustedTransfer) continue;
-      result.push(new TransactionItem(nestedTransaction));
-    }
-    return result;
+        return !onlyTrusted || this.isTrustedTransfer(nestedTransaction);
+      })
```

Comment:
nitpick:

```typescript
.filter((nestedTransaction): boolean => {
  // We are interested in transfers that:
  // - Have value and:
  // - If onlyTrusted is true then it should be a trusted transfer
  // - If onlyTrusted is false then any transfer is valid
  return (
    this.isTransferWithValue(nestedTransaction) &&
    (!onlyTrusted || this.isTrustedTransfer(nestedTransaction))
  );
})
```

Reply:
Included in 15489f36e5464501c0356a4652eae757dd04c396.